spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/support/converter/SqsHeaderMapper.java

```diff
@@ -181,11 +181,21 @@ private Object getValue(Map.Entry<String, MessageAttributeValue> entry) {
 		MessageAttributeValue value = entry.getValue();
 		String dataType = value.dataType();
 		Assert.notNull(dataType, "dataType must not be null");
-		return MessageAttributeDataTypes.STRING.equals(dataType)
-			? value.stringValue()
-			: MessageAttributeDataTypes.BINARY.equals(dataType)
-				? value.binaryValue()
-				: getNumberValue(value);
+
+		if (dataType.contains(".")) {
+			dataType = dataType.substring(0, dataType.indexOf('.'));
+		}
+
+		switch (dataType) {
```

Comment:
Given we're in JDK 17, IntelliJ suggested this instead:
```java
		return switch (dataType) {
			case MessageAttributeDataTypes.NUMBER -> getNumberValue(value);
			case MessageAttributeDataTypes.BINARY -> value.binaryValue();
			default -> value.stringValue();
		};
```

Had no idea this was possible, but sure looks good. WDYT?

Reply:
Updated to use the new switch statements.